pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -223,7 +223,7 @@ async fn download_object<'a>(
                 // There's chunks_vectored() on the stream.
                 let (bytes_amount, destination_file) = async {
                     let mut buffered =
-                        owned_buffers_io::write::BufferedWriter::<IoBufferMut, _>::new(
+                        owned_buffers_io::write::BufferedWriter::<IoBufferMut, _>::new_no_gate_guard(
```

Comment:
Why is it safe to not use the gate guard here?

What makes this place special so it can't happen that we leak the flush task?

I don't think this place is special